(feat): toggling extensions

hjbdev opened the pull request:
This PR adds functionality for toggling extensions in the php.ini with `pvm extensions <enable|disable> <ext>`

Older installs may need to run `pvm use <version>` once after upgrading, because the active version is now stored in `~/.pvm/version`

hjbdev committed:
wip toggling extensions

HerrSammyDE commented:
When can we expect the update?

hjbdev commented:
still needs some work from my end, will try and get it done in January

Lotti commented:
I'd like to contribute to this feature (probably somewhere in February, hoping to have some spare hours at work) by introducing download and setup of builded dll from pecl registry.
Unfortunately the pecl registry website is a bit tedious to navigate and parse so it will be require lots of http calls if I will not find a better interface for determine what is the latest release of extension X compatible with php Y.

eydun commented:
Looking forward to this feature.

hjbdev committed:
Merge remote-tracking branch 'origin/master' into extension-toggling

# Conflicts:
#	commands/list.go
#	commands/use.go
#	common/helpers.go

hjbdev committed:
Refactor extension handling and improve error reporting in Extensions command

hjbdev committed:
Harden extension toggle updates

hjbdev committed:
Add extension inventory listing

hjbdev committed:
Support toggling Zend extensions

hjbdev committed:
Add php.ini comment parsing regression test

hjbdev committed:
Add extensions coverage and docs

Cover duplicate directive and active-version metadata cases, and document the new pvm extensions workflow.
